server/middleware/patchable.coffee

```diff
@@ -54,3 +63,97 @@ module.exports =
+
+    # normalize the delta because in tests, changes to patches would sneak in and cause false positives
+    # TODO: Figure out a better system. Perhaps submit a series of paths? I18N Edit Views already use them for their rows.
+    originalDelta = req.body.delta
+    originalTarget = target.toObject()
+    newTargetAttrs = _.cloneDeep(target.toObject(), (value) ->
+      return value if value instanceof mongoose.Types.ObjectId
+      return value if value instanceof Date
+      return undefined
+    )
+    jsondiffpatch.patch(newTargetAttrs, originalDelta)
+    normalizedDelta = jsondiffpatch.diff(originalTarget, newTargetAttrs)
+    normalizedDelta = _.pick(normalizedDelta, _.keys(originalDelta))
+
+    # decide whether the patch should be auto-accepted, or left 'pending' for an admin or artisan to review
+    reasonNotAutoAccepted = undefined
+    validation = tv4.validateMultiple(newTargetAttrs, Model.jsonSchema)
+    if not validation.valid
+      reasonNotAutoAccepted = 'Did not pass json schema.'
+    else if not isJustFillingTranslations(normalizedDelta)
+      reasonNotAutoAccepted = 'Adding to existing translations.'
+      
+    else
+      # save changes directly to the target, whether versioned or not, and send out notifications
+      if Model.schema.uses_coco_versions
+        newVersion = database.initDoc(req, Model)
+        initNewVersion(newVersion, target)
+        newVersionAttrs = newVersion.toObject()
+        jsondiffpatch.patch(newVersionAttrs, normalizedDelta)
+        newVersion.set(newVersionAttrs)
+        newVersion.set('commitMessage', req.body.commitMessage)
+        major = target.get('version.major')
+        updateI18NCoverage(newVersion)
+        yield saveNewVersion(newVersion, major)
+        target = newVersion
+      else
+        target.set(newTargetAttrs)
+        updateI18NCoverage(target)
+        yield target.save()
+      notifyChangesMadeToDoc(req, target)
+
+    # create, validate and save the patch
+    if Model.schema.uses_coco_versions
+      patchTarget = {
+        collection: collectionName
+        id: target._id
+        original: target._id
+        version: _.pick(target.get('version'), 'major', 'minor')
+      }
+    else
+      patchTarget = {
+        collection: collectionName
+        id: target._id
+        original: target._id
+      }
+
+    patch = new Patch()
+    patch.set({
+      delta: normalizedDelta
+      commitMessage: req.body.commitMessage
+      target: patchTarget
+      creator: req.user._id
+      status: if reasonNotAutoAccepted then 'pending' else 'accepted'
+      created: new Date().toISOString()
+      reasonNotAutoAccepted: reasonNotAutoAccepted
+    })
+    database.validateDoc(patch)
+
+    # add this patch to the denormalized list of patches on the target
+    if reasonNotAutoAccepted
+      yield target.update({ $addToSet: { patches: patch._id }})
+      patches = target.get('patches') or []
+      patches.push patch._id
+      target.set({patches})
```

Comment:
It looks like this does 2 different ways of adding to `target.patches` — was one of these intended to be taken out?